device-connectors/src/testflinger_device_connectors/devices/zapper_iot/__init__.py

```diff
@@ -45,6 +45,10 @@ def _validate_configuration(
             "test_password", "ubuntu"
         )
 
+        ubuntu_sso_email = self.job_data["provision_data"].get(
+            "ubuntu_sso_email", ""
+        )
+
         provisioning_data = {
             "username": username,
```

Comment:
What about we already assign it here?
```suggestion
            "username": username if not ubuntu_sso_email else ubuntu_sso_email,
```

Reply:
Agree, it will be clearer.